src/components/dialogs/parameters/dynamicsimulation/curve/dialog/equipment-filter.jsx

```diff
@@ -259,17 +248,18 @@ const EquipmentFilter = forwardRef(
             selectedNominalVoltages,
         ]);
 
+        // fetching filtered equipments
         useEffect(() => {
             let ignore = false;
-            if (gridReady && countriesFilterReady) {
-                equipmentsRef.current.api.showLoadingOverlay();
+            if (gridReady) {
+                equipmentsRef?.current?.api?.showLoadingOverlay();
```

Comment:
If AG Grid is ready, why these `?`

Reply:
YES! not needed using ? before call fetch..

In order to make the code more clear and respect the single trust source principle, I have refactorized the callback onGridReady() to set the grid api instead of using a flag gridReady, see https://github.com/gridsuite/gridstudy-app/pull/1836/files#diff-4534b7ceb143990bde6053ad363afdf1abb64fc68ccf95830c93a3ed1041a205R302

DONE in https://github.com/gridsuite/gridstudy-app/commit/0bef9a2e0bd4c319f0e8f31aa7ea0cd139c91dc1

Reply:
I'm not sure about this refactoring.
That's not how AG Grid recommend it.
https://ag-grid.com/react-data-grid/grid-interface/#grid-api
You're sure of what you did ? If yes you can keep it like that, but well... what is the `api` ref change ? You will get null error.
My remark was just on the lot of `?` that were useless because the AG Grid API was ready.

Reply:
Reverted by using ref.current?.api as previously... add only ? to avoid null error when click quickly several options then close dialogue immediately..

DONE

Reply:
AS we said can you add a comment to explain why we need this ?

Reply:
DONE in https://github.com/gridsuite/gridstudy-app/pull/1836/commits/60a80dcec4d6ecd0438dc480b8d50b1971730e25